legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/operational/errorManagement/ErrorCategory.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.shared.core.operational.errorManagement;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * User friendly error categories
+ */
+public enum ErrorCategory
+{
+    @JsonProperty("UserAuthenticationError")
+    USER_AUTHENTICATION_ERROR,
+    @JsonProperty("UserExecutionError")
+    USER_EXECUTION_ERROR,
+    @JsonProperty("InternalServerError")
+    INTERNAL_SERVER_ERROR,
+    @JsonProperty("ServerExecutionError")
+    SERVER_EXECUTION_ERROR,
+    @JsonProperty("OtherError")
+    OTHER_ERROR,
+    UNKNOWN_ERROR,
+}
```

Comment:
Unknown is for when an exception comes in that cannot be categorised to any of the other five categories:
- No category is passed when throwing the exception
- No matching method (using the error data file) is a hit on the incoming exception
Unknown error is also the default value for an EngineException unless specified otherwise.

Other error is for exceptions that are not caused by the user or server, which currently includes:
- Connection and timeout related exceptions
- Test related exceptions